src/parser/jobs/rpr/modules/Gauge.tsx

```diff
@@ -0,0 +1,183 @@
+		deterministic: false,
+	}))
+
+	override initialise() {
+		super.initialise()
+
+		const playerFilter = filter<Event>().source(this.parser.actor.id)
+
+		// soul gauge
+		this.addEventHook(playerFilter.type('action').action(this.data.matchActionId(SOUL_GENERATORS)), this.onSoulGeneration)
+		this.addEventHook(playerFilter.type('action').action(this.data.matchActionId(SOUL_CONSUMERS)), this.onSoulConsumption)
+
+		// shroud gauge
+		this.addEventHook(playerFilter.type('action').action(this.data.matchActionId(SHROUD_GENERATORS)), this.onShroudGeneration)
+		this.addEventHook(playerFilter.type('action').action(this.data.matchActionId(SHROUD_CONSUMERS)), this.onShroudConsumption)
+
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	private onSoulGeneration(event: Events['action']) {
+		const generatedAmount = this.soulGeneratorModifiers.get(event.action)
+		if (generatedAmount) {
+			this.soulGauge.generate(generatedAmount)
+		}
+	}
+
+	private onSoulConsumption(event: Events['action']) {
+		const spentAmount = this.soulConsumptionModifiers.get(event.action)
+		if (spentAmount) {
+			this.soulGauge.spend(spentAmount)
+		}
+	}
+
+	private onShroudGeneration(event: Events['action']) {
+		const generated = this.shroudGeneratorModifiers.get(event.action)
+		if (generated) {
+			this.shroudGauge.generate(generated)
+		}
+	}
+
+	private onShroudConsumption(event: Events['action']) {
+		const spent = this.shroudConsumerModifiers.get(event.action)
+		if (spent) {
+			this.shroudGauge.spend(spent)
+		}
+	}
+
+	private onComplete() {
+		this.suggestions.add(new TieredSuggestion({
+			icon: this.data.actions.SOUL_SLICE.icon,
+			content: <Trans id="rpr.gauge.suggestions.soul-gauge-overcap.content">
+				Try to not overcap Soul gauge since it's required for shroud generation for your highest damaging skills
+				via <DataLink action="ENSHROUD"/>.
+				Pay special attention when using <DataLink action="SOUL_SLICE"/> or <DataLink action="SOUL_SCYTHE"/> as
+				they increase the gauge by <b>50</b>, make sure you have <b>50</b> or less in your gauge before using.
+			</Trans>,
+			tiers: {
+				10: SEVERITY.MINOR,
+				30: SEVERITY.MEDIUM,
+				50: SEVERITY.MAJOR,
+			},
+			value: this.soulGauge.overCap,
+			why: <Trans id="rpr.gauge.suggestions.soul-gauge-overcap.why">
+				You lost {this.soulGauge.overCap} Soul due to overcapped gauge.
+			</Trans>,
+		}))
+
+		this.suggestions.add(new TieredSuggestion({
+			icon: this.data.actions.ENSHROUD.icon,
+			content: <Trans id="rpr.gauge.suggestions.shroud-gauge-overcap.content">
+				Try to not overcap Shroud gauge because it can lead to less usages of <DataLink action="ENSHROUD"/> over
+				the course of the fight.
+			</Trans>,
+			tiers: {
+				10: SEVERITY.MINOR,
+				30: SEVERITY.MEDIUM,
+				50: SEVERITY.MAJOR,
+			},
+			value: this.shroudGauge.overCap,
+			why: <Trans id="rpr.gauge.suggestions.shroud-gauge-overcap.why">
+				You lost {this.shroudGauge.overCap} Shroud due to overcapped gauge.
```

Comment:
Food for thought: it might be more beneficial to users to translate this into lost uses of Enshroud, rather than a raw "how much gauge you lost" number. I've found that to be easier to understand. Won't block on this one if you'd rather keep it as-is though. Same deal with the soul gauge overcap suggestion too.

Reply:
Any suggestion on what would be said in case of not a single lost use, like you lost 40 shroud gauge?

Reply:
Chiming in:
I personally would go for something more along the lines of
"You lost {this.shroudGauge.overCap} Shroud gauge due to overcapping, which cost you {this.shroudGauge.overCap / SHROUD_SPEND_AMOUNT} uses of Enshroud."